### PR TITLE
Change `debug!` to `trace!` to reduce logging noise

### LINKED ISSUE #2282: esp-wifi: Reduce logging level for normal timer operations.
When building with log level `Debug`, the logs quickly get polluted by the following log lines. I believe, since they are normal operations, they should be degraded to trace level:

_*Note: I may be forgetting others, but these are the general ones I'm seeing_

https://github.com/esp-rs/esp-hal/blob/d3f03042f317f6e80c42c407015a0b452723c106/esp-wifi/src/compat/timer_compat.rs#L56-L59

https://github.com/esp-rs/esp-hal/blob/d3f03042f317f6e80c42c407015a0b452723c106/esp-wifi/src/compat/timer_compat.rs#L76

https://github.com/esp-rs/esp-hal/blob/d3f03042f317f6e80c42c407015a0b452723c106/esp-wifi/src/compat/timer_compat.rs#L109-L112

https://github.com/esp-rs/esp-hal/blob/d3f03042f317f6e80c42c407015a0b452723c106/esp-wifi/src/tasks.rs#L27

Example of the current logging:
```
DEBUG - timer_disarm 3fccf5e0
DEBUG - timer_disarm 3fccf5e0
DEBUG - timer_setfn 3fcce364 0x420f6b3c 0x0
0x420f6b3c - ieee80211_timer_process
    at ??:??
DEBUG - timer_setfn 3fcce378 0x420f6b3c 0x1
0x420f6b3c - ieee80211_timer_process
    at ??:??
DEBUG - timer_disarm 3fccf5e0
DEBUG - timer_disarm 3fccdb00 not found
DEBUG - timer_done 3fccdb00 not found
DEBUG - timer_setfn 3fccdb00 0x420e4260 0x0
DEBUG - timer_arm_us 3fccdb00 current: 1012271 ticks: 103000 repeat: false
DEBUG - timer_disarm 3fccf5e0
DEBUG - timer_disarm 3fccf5e0
DEBUG - timer_disarm 3fccf5e0
INFO - [AP] Wifi started
DEBUG - timer is due.... 3fccdb00
DEBUG - timer_disarm 3fccdb00
DEBUG - timer_arm_us 3fccdb00 current: 1135202 ticks: 83000 repeat: false
DEBUG - timer is due.... 3fccdb00
DEBUG - timer_disarm 3fccdb00
DEBUG - timer_arm_us 3fccdb00 current: 1235627 ticks: 85000 repeat: false
DEBUG - timer is due.... 3fccdb00
DEBUG - timer_disarm 3fccdb00
DEBUG - timer_arm_us 3fccdb00 current: 1334635 ticks: 88000 repeat: false
DEBUG - timer is due.... 3fccdb00

```

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [ ] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [ ] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description
Fixes #2282 - most of these are just noise and irrelevant for users

This doesn't deserve a changelog entry

#### Testing
Examples still work

### CODE REVIEW

**Review by jessebraham — APPROVED** [review]
Glad I saw this, was just going to start working on this myself 😁 Thanks!

File: esp-wifi/src/compat/timer_compat.rs [hunk 1]
```diff
@@ -65,18 +68,18 @@ pub fn compat_timer_arm_us(ets_timer: *mut ets_timer, us: u32, repeat: bool) {
             timer.active = true;
             timer.periodic = repeat;
         } else {
-            debug!("timer_arm_us {:x} not found", ets_timer as usize);
+            trace!("timer_arm_us {:x} not found", ets_timer as usize);
```
Comment (bugadani) [inline esp-wifi/src/compat/timer_compat.rs:71]: This probably should remain at a higher level

File: esp-wifi/src/compat/timer_compat.rs [hunk 2]
```diff
@@ -65,18 +68,18 @@ pub fn compat_timer_arm_us(ets_timer: *mut ets_timer, us: u32, repeat: bool) {
...
 
 pub fn compat_timer_disarm(ets_timer: *mut ets_timer) {
     critical_section::with(|_| unsafe {
         if let Some(timer) = TIMERS.iter_mut().find(|t| t.ets_timer == ets_timer) {
-            debug!("timer_disarm {:x}", timer.id());
+            trace!("timer_disarm {:x}", timer.id());
             timer.active = false;
         } else {
-            debug!("timer_disarm {:x} not found", ets_timer as usize);
+            trace!("timer_disarm {:x} not found", ets_timer as usize);
```
Comment (bugadani) [inline esp-wifi/src/compat/timer_compat.rs:82]: This one, too

File: esp-wifi/src/compat/timer_compat.rs [hunk 3]
```diff
@@ -88,15 +91,15 @@ pub fn compat_timer_done(ets_timer: *mut ets_timer) {
...
+            trace!("timer_done {:x}", timer.id());
             timer.active = false;
 
             (*ets_timer).priv_ = core::ptr::null_mut();
             (*ets_timer).expire = 0;
 
             TIMERS.swap_remove(idx);
         } else {
-            debug!("timer_done {:x} not found", ets_timer as usize);
+            trace!("timer_done {:x} not found", ets_timer as usize);
```
Comment (bugadani) [inline esp-wifi/src/compat/timer_compat.rs:102]: Same

File: esp-wifi/src/compat/timer_compat.rs (hunk 1 above)
Comment (bjoernQ) [reply esp-wifi/src/compat/timer_compat.rs:71]: these are all quite frequent (with BLE at least) currently while things work fine -maybe they should be kept as debug and investigated

Comment (bjoernQ) [reply esp-wifi/src/compat/timer_compat.rs:71]: I _guess_ it's fine if they call `disarm` or `done` on a timer we don't know (yet) about
I'm not too sure if it's the same for `arm`? In theory if they never did `setfn` we cannot/don't need to do much - or maybe we should just initialize the timer in `arm` anyway with a no-op callback

**Review by bugadani — APPROVED** [review]

**Review by JurajSadel — APPROVED** [review]
Thanks!

**Pull request merged**